Enable Trimming on FileProviders.Embedded

eerhardt opened the pull request:
The FileProviders.Embedded assembly wasn't marked as trimmable, and wasn't getting analyzers run on it. Enable the analyzers and fix the warnings.

Clean up items:
- Capture the result of Assembly.Location in a variable because it allocates a string every time it is called.
- Remove the TrimmingAttributes.cs file from AspNetCore.Identity because the assembly only builds for DefaultNetCoreTargetFramework.

eerhardt committed:
Enable Trimming on FileProviders.Embedded

The FileProviders.Embedded assembly wasn't marked as trimmable, and wasn't getting analyzers run on it. Enable the analyzers and fix the warnings.

Clean up items:
- Capture the result of Assembly.Location in a variable because it allocates a string every time it is called.
- Remove the TrimmingAttributes.cs file from AspNetCore.Identity because the assembly only builds for DefaultNetCoreTargetFramework.

captainsafia commented on src/FileProviders/Embedded/src/Microsoft.Extensions.FileProviders.Embedded.csproj:
```diff
@@ -12,6 +12,7 @@
     <PackageTags>files;filesystem</PackageTags>
     <NoPackageAnalysis>true</NoPackageAnalysis>
     <Nullable>enable</Nullable>
+    <IsTrimmable>true</IsTrimmable>
```
Based on the [docs on configuring trimming](https://github.com/dotnet/aspnetcore/blob/bec278eabea54f63da15e10e654bdfa4168a2479/docs/Trimming.md#trim-an-aspnet-core-assembly) I believe that we should also see an update to [eng/TrimmableProjects.props](https://github.com/dotnet/aspnetcore/blob/bec278eabea54f63da15e10e654bdfa4168a2479/eng/TrimmableProjects.props) following this change.

captainsafia commented on src/Identity/Core/src/Microsoft.AspNetCore.Identity.csproj:
```diff
@@ -13,9 +13,6 @@
     <Reference Include="Microsoft.AspNetCore.Authentication.Cookies" />
     <Reference Include="Microsoft.Extensions.Identity.Core" />
-
-    <Compile Include="$(SharedSourceRoot)TrimmingAttributes.cs" LinkBase="Shared"
```
Why is this being removed?

eerhardt committed:
Update TrimmableProjects.

Also fix CodeGen.proj to use Environment.NewLine, so it doesn't make unnecessary diffs on Windows machines.

amcasey reviewed (approved):
Seems reasonable.

eerhardt committed:
Update Tools.slnf
